crates/broker/src/order_picker.rs

```diff
@@ -750,6 +751,12 @@ where
                             current_capacity = new_capacity;
                         }
                     }
+                    _ = cancel_token.cancelled() => {
+                        tracing::info!("Order picker received cancellation, shutting down gracefully");
+                        // Abort remaining tasks
+                        tasks.abort_all();
```

Comment:
Related to the above comment, here is where I'd expect us to cancel any ongoing executor runs (assuming thats possible) if we went with the "task owns its own cleanup" approach

Reply:
I was considering this, but was under the assumption this wouldn't affect in-progress exec jobs. I thought those wouldn't be cancelled by our current session cancellation logic. Let me double check